contextualized/regression/datasets.py

```diff
@@ -135,3 +135,14 @@ def __init__(self, dataset):
 
     def __iter__(self):
         return iter(self.dataset)
+
+
+def singletask_worker_init_fn(worker_id):
+    worker_info = torch.utils.data.get_worker_info()
+    dataset = worker_info.dataset
+    split_size = len(dataset.dataset.C) // worker_info.num_workers  # divide equally among workers
```

Comment:
Are there any memory issues with using `dataset.dataset.C`? Does the dataset get copied for each worker?

Reply:
This is a good point since each worker gets its own copy of the dataset. This will be investigated further

Reply:
I looked into this, TL;DR this isn't an issue as long as the data inputs are pandas or numpy arrays, which we already require.

Torch docs for this: https://pytorch.org/docs/stable/data.html
Master issue for dataset duplication with multithreading: https://github.com/pytorch/pytorch/issues/13246

Solutions
1. (Recommended by Torch): use non-refcounted data representations (pandas or numpy).
2. Keep data in cold storage (each datapoint in a separate file) and split a list of filenames among workers. This is similar in principle to 1 (a single dataset shared by all workers) but slower. We should only use this for massive datasets, where the data cannot be loaded into memory all at once.